0.1.Copy data from Excel codes.R

```diff
@@ -1,79 +1,75 @@
 
 # calculate ibm
-subject_infor$ibm <- as.numeric(subject_infor$LBM)/as.numeric(subject_infor$WT)
-subject_infor <- as.data.frame(subject_infor)
+dat0$ibm <- with(dat0,LBM/WT)
 
 # save data
-saveRDS(subject_infor, file = "E:/2021-03-26 Bulk Fat RNAseq/Subjects basic infor from Clamp.Rds")
-write.csv(subject_infor, file = "E:/2021-03-26 Bulk Fat RNAseq/Subjects basic infor from Clamp.csv", row.names = FALSE)
-write.xlsx(subject_infor, file = "E:/2021-03-26 Bulk Fat RNAseq/Subjects basic infor from Clamp.xlsx", row.names = FALSE)
+#+ save_data
+saveRDS(dat0,file='Subjects basic infor from Clamp.Rds')
+write.csv(dat0, file = "Subjects basic infor from Clamp.csv", row.names = FALSE)
+write.xlsx2(dat0, file = "Subjects basic infor from Clamp.xlsx", row.names = FALSE)
 
 
-## Import clamp data 
-clamp <- read.xlsx2("E:/2021-03-26 Bulk Fat RNAseq/Subjects with Clamp Infor.xlsx", sheetName = "Subjects", header=TRUE)
+## Import clamp data
+#+ clamp_data
+clamp <- read_xlsx(inputdata['subjects'],sheet='Subjects');
 
 # Organize FFA parameters
 groups = c( "FFA_basal", "FFA_20", "FFA_80")
-Corr.Result_p <- as.data.frame(clamp[, groups])
-rownames(Corr.Result_p) <- clamp$Subject_ID
-for ( col in colnames(Corr.Result_p)) { Corr.Result_p <- Corr.Result_p[!grepl("ERROR", Corr.Result_p[, col]),] }
-
-# calculate AUC
-library(bayestestR)
-for (g in 1:nrow(Corr.Result_p)) {
-  Corr.Result_p[g, "AUC"] <- area_under_curve(c(0,20,80), as.numeric(Corr.Result_p[g, groups]), method = "trapezoid")
-  Corr.Result_p[g, "Total"] <- (as.numeric(Corr.Result_p[g, groups[1]]) * (80-0)) 
-}
-Corr.Result_p[["FFA.revAUC"]] <- Corr.Result_p[["Total"]] - Corr.Result_p[["AUC"]]
-hist(Corr.Result_p[["FFA.revAUC"]])
-Corr.Result_p[["Subject_ID"]] <- rownames(Corr.Result_p)
+#+ AUC
+clamp$AUC <- apply(clamp,1,function(xx) area_under_curve(c(0,20,80),as.numeric(xx[groups])));
+clamp$Total <- clamp[[groups[1]]]*80;
+clamp$FFA.revAUC <- with(clamp,Total - AUC);
 
-# add FFA.revAUC to clamp data
-clamp <- merge(clamp, Corr.Result_p[, c("Subject_ID", "FFA.revAUC")], by = "Subject_ID", all.x = TRUE)
 
 ## Combine all information
-all.infor <- merge(subject_infor, clamp, by= intersect(colnames(subject_infor), colnames(clamp)) , all = TRUE)
+all.infor <- merge(dat0, clamp, by= intersect(colnames(dat0), colnames(clamp)) , all = TRUE)
 
 # save data
-saveRDS(all.infor, file = "E:/2021-03-26 Bulk Fat RNAseq/Subjects with Clamp Infor and All Infor.RDS")
-write.csv(all.infor, file = "E:/2021-03-26 Bulk Fat RNAseq/Subjects with Clamp Infor and All Infor.csv", row.names = FALSE)
-write.xlsx(all.infor, "E:/2021-03-26 Bulk Fat RNAseq/Subjects with Clamp Infor and All Infor.xlsx", row.names = FALSE)
+saveRDS(all.infor, file = "Subjects with Clamp Infor and All Infor.RDS")
+write.csv(all.infor, file = "Subjects with Clamp Infor and All Infor.csv", row.names = FALSE)
+write.xlsx(all.infor, "Subjects with Clamp Infor and All Infor.xlsx", row.names = FALSE)

```

Comment:
As above, saving to project directory to avoid making assumptions about what drives exist.